jwst/tweakreg/tweakreg_step.py

```diff
@@ -455,17 +464,58 @@ def process(self, input):
-                    expand_refcat=False,
-                    minobj=self.abs_minobj,
-                    match=xyxymatch_gaia,
-                    fitgeom=self.abs_fitgeometry,
-                    nclip=self.abs_nclip,
-                    sigma=(self.abs_sigma, 'rmse')
-                )
+                try:
+                    align_wcs(
+                        imcats,
+                        refcat=ref_cat,
+                        enforce_user_order=True,
+                        expand_refcat=False,
+                        minobj=self.abs_minobj,
+                        match=xyxymatch_gaia,
+                        fitgeom=self.abs_fitgeometry,
+                        nclip=self.abs_nclip,
+                        sigma=(self.abs_sigma, 'rmse')
+                    )
+                except ValueError as e:
+                    msg = e.args[0]
+                    if (msg == "Too few input images (or groups of images) with "
+                            "non-empty catalogs."):
+                        # we need at least two exposures to perform image alignment
+                        self.log.warning(msg)
+                        self.log.warning(
+                            "At least one exposure is required to align images "
+                            "to an absolute reference catalog. Alignment to an "
+                            "absolute reference catalog will not be performed."
+                        )
+                        if not rel_ok:
+                            self.log.warning("Nothing to do. Skipping 'TweakRegStep'...")
+                            for model in images:
+                                model.meta.cal_step.tweakreg = "SKIPPED"
+                            self.skip = True
+                            return images
+                    else:
+                        raise e
+
+                except RuntimeError as e:
+                    msg = e.args[0]
+                    if msg.startswith("Number of output coordinates exceeded allocation"):
+                        # we need at least two exposures to perform image alignment
+                        self.log.error(msg)
+                        self.log.error(
+                            "Multiple sources within specified tolerance "
+                            "matched to a single reference source. Try to "
+                            "adjust 'tolerance' and/or 'separation' parameters."
+                            "Alignment to an absolute reference catalog will "
+                            "not be performed."
+                        )
+                        if not rel_ok:
+                            self.log.warning("Skipping 'TweakRegStep'...")
+                            self.skip = True
+                            for model in images:
+                                model.meta.cal_step.tweakreg = "SKIPPED"
+                            return images
+                    else:
+                        raise e
```

Comment:
Just a suggestion: I think that the `except` block can be refactored to avoid duplicate code.

The only effective difference between handling a `ValueError` or a `RuntimeError` is in the log type and message, which means we could have one single `except` block for both cases wherein an if statement can be used to pick the correct log message. Alternatively, the error handling could be implemented as a separate function.

Reply:
There's a follow up PR which refactors tweakreg. I'm inclined to leave the refactoring to that PR.

Reply:
Actually, I do not. I prefer using `except ErrorType` to have different actions instead of catching all in one and then analyzing error message or log or whatever. It is cleanest in my opinion. Just because issues log message or action taken for each exception are similar, I do not think that minimal savings in code lines justifies reduced clarity.

Reply:
OK, so I'll declare this as resolved.